### PR TITLE
Add top ports to agent nmap

### PR DESCRIPTION
**Pull request by BlueSquare1**

### CODE REVIEW

File: agent/nmap_options.py [hunk 1]
```diff
@@ -45,6 +45,7 @@ class NmapOptions:
     dns_resolution: bool = True
     dns_servers: List[str] | None = None
     ports: Optional[str] | None = None
+    top_ports: Optional[int] | None = None
```
Comment (amine3) [inline agent/nmap_options.py:48]: int | None = None What does the Optional do here?

File: agent/nmap_options.py [hunk 2]
```diff
@@ -97,6 +98,8 @@ def _set_ports_option(self) -> List[str]:
         """Appends the ports option to the list of nmap options."""
         if self.fast_mode is True:
             return ["-F"]
+        elif self.top_ports is not None:
+            return ["--top-ports", str(self.top_ports)]
```
Comment (amine3) [inline agent/nmap_options.py:102]: if top_ports has the same syntax as ports, ["--top-ports", self.top_ports] would work no?

**Review by amine3 — CHANGES_REQUESTED** [review]

File: tests/configs/top_ports.yaml [hunk 3]
```diff
@@ -0,0 +1,128 @@
+kind: Agent
```
Comment (3asm) [inline tests/configs/top_ports.yaml:1]: What is this file?

File: tests/configs/fast_mode.yaml [hunk 4]
```diff
@@ -0,0 +1,127 @@
+kind: Agent
```
Comment (3asm) [inline tests/configs/fast_mode.yaml:1]: What is this file?

File: tests/configs/all_ports.yaml [hunk 5]
```diff
@@ -0,0 +1,127 @@
+kind: Agent
```
Comment (3asm) [inline tests/configs/all_ports.yaml:1]: What is this file?

File: agent/nmap_wrapper.py [hunk 6]
```diff
@@ -39,7 +39,7 @@ def __init__(self, options: nmap_options.NmapOptions) -> None:
         """
         self._options = options
 
-    def _construct_command_host(self, host: str, mask: int) -> List[str]:
+    def construct_command_host(self, host: str, mask: int) -> List[str]:
```
Comment (3asm) [inline agent/nmap_wrapper.py:42]: Why has this become public?

File: tests/nmap_wrapper_test.py [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+    command = client.construct_command_host("127.0.0.1", 24)
+
+    assert "-F" in command
+    assert "--top-ports" not in command
+    assert "-p" not in command
+
+
+@pytest.mark.parametrize(
+    "nmap_parametrized_agent",
+    [["top_ports.yaml"]],
```
Comment (3asm) [inline tests/nmap_wrapper_test.py:40]: This is not how we test the config, because those files are not maintainable, you should pass the config instead.

**Review by 3asm — CHANGES_REQUESTED** [review]

File: agent/nmap_wrapper.py (hunk 6 above)
Comment (BlueSquare1) [reply agent/nmap_wrapper.py:42]: it's part of the test.

File: tests/conftest.py [hunk 8]
```diff
@@ -280,3 +280,88 @@ def ipv6_msg_above_limit() -> message.Message:
...
+) -> nmap_agent.NmapAgent:
+    """Fixture of the Nmap Agent to be used for testing purposes."""
+    del agent_persist_mock
+    with (pathlib.Path(__file__).parent.parent / "ostorlab.yaml").open() as yaml_o:
+        definition = agent_definitions.AgentDefinition.from_yaml(yaml_o)
+        settings = runtime_definitions.AgentSettings(
+            key="agent/ostorlab/nmap_agent",
+            bus_url="NA",
+            bus_exchange_topic="NA",
+            args=[],
```
Comment (3asm) [inline tests/conftest.py:298]: I would set it explicitly to avoid changes in default settings.

File: tests/nmap_wrapper_test.py [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+        version_detection=args.get("version_info", False),
+    )
+    client = nmap_wrapper.NmapWrapper(options)
+
+    command = client.construct_command_host("127.0.0.1", 24)
+
+    assert "--top-ports" in command
+    assert "420" in command
+    assert "-F" not in command
+    assert "-p" not in command
```
Comment (3asm) [inline tests/nmap_wrapper_test.py:54]: assert on the full command.

File: tests/nmap_wrapper_test.py [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+        version_detection=args.get("version_info", False),
+    )
+    client = nmap_wrapper.NmapWrapper(options)
+
+    command = client.construct_command_host("127.0.0.1", 24)
+
+    assert "-p" in command
+    assert "0-65535" in command
+    assert "-F" not in command
+    assert "--top-ports" not in command
```
Comment (3asm) [inline tests/nmap_wrapper_test.py:79]: Assert on the full command.

**Review by 3asm — APPROVED** [review]

**Pull request merged**